hazelcast/include/hazelcast/client/cluster/impl/VectorClock.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef HAZELCAST_CLIENT_CLUSTER_IMPL_VECTORCLOCK_H
+#define HAZELCAST_CLIENT_CLUSTER_IMPL_VECTORCLOCK_H
+
+#include <string>
+#include <stdint.h>
+
+#include "hazelcast/util/SynchronizedMap.h"
+#include "hazelcast/client/serialization/IdentifiedDataSerializable.h"
+
+namespace hazelcast {
+    namespace client {
+        namespace cluster {
+            namespace impl {
+                /**
+                 * Vector clock consisting of distinct replica logical clocks.
+                 * <p>
+                 * See https://en.wikipedia.org/wiki/Vector_clock
+                 * The vector clock may be read from different thread but concurrent
+                 * updates must be synchronized externally. There is no guarantee for
+                 * concurrent updates.
+                 */
+                class VectorClock : public serialization::IdentifiedDataSerializable {
```

Comment:
We don't send this to remote so This does not need to be Serilazable, right ?

Reply:
Removed it.